### PR TITLE
various small host fixes

### PR DESCRIPTION
**Pull request by KKopyscinski**

### CODE REVIEW

File: nimble/host/mesh/src/testing.c [hunk 1]
```diff
@@ -171,7 +171,9 @@ void bt_test_print_credentials(void)
 		console_printf("Friend cred: %d\n", i);
 		console_printf("\tNetKeyIdx: %04x\n",
 			           sub[i].net_idx);
-		console_printf("\tNID: %02x\n", subnet->keys->msg.nid);
+		if (subnet) {
+            console_printf("\tNID: %02x\n", subnet->keys->msg.nid);
+        }
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/testing.c:176]: What about the next line? And indentation.

File: nimble/host/mesh/src/friend.c [hunk 2]
```diff
@@ -1224,7 +1224,6 @@ static void friend_timeout(struct ble_npl_event *work)
 	}
 
 	/* Clear the flag we use for segment tracking */
-	BT_MESH_ADV(frnd->last)->flags &= ~NET_BUF_FRAGS;
 	BT_MESH_ADV(frnd->last)->flags = 0;
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/friend.c:1227]: This is a typo. The second line should be `frags` and you should set it to NULL (this is how it looks in Zephyr). And we should not erase the first line. We need to double-check this.

File: nimble/host/mesh/src/adv.c [hunk 3]
```diff
@@ -232,8 +232,8 @@ struct os_mbuf *bt_mesh_adv_create_from_pool(struct os_mbuf_pool *pool,
 	adv->ref_cnt = 1;
 	ble_npl_event_set_arg(&adv->ev, buf);
 
-	return buf;
 	os_mbuf_free_chain(buf);
+    return buf;
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/adv.c:236]: Indentation.

**Review by michal-narajowski — CHANGES_REQUESTED** [review]

File: nimble/host/mesh/src/testing.c (hunk 1 above)
Comment (KKopyscinski) [reply nimble/host/mesh/src/testing.c]: done

File: nimble/host/mesh/src/friend.c (hunk 2 above)
Comment (KKopyscinski) [reply nimble/host/mesh/src/friend.c]: I checked this method with Zephyr and that typo and value is the only difference. Synced with Zephyr

File: nimble/host/mesh/src/adv.c (hunk 3 above)
Comment (KKopyscinski) [reply nimble/host/mesh/src/adv.c]: done

File: nimble/host/mesh/src/friend.c (hunk 2 above)
Comment (KKopyscinski) [reply nimble/host/mesh/src/friend.c]: As disgussed offline, this requires bigger fix. I'll remove this from this PR

File: nimble/controller/src/ble_ll_scan.c [hunk 4]
```diff
@@ -3081,7 +3081,7 @@ ble_ll_scan_rx_pkt_in_on_aux(uint8_t pdu_type, struct os_mbuf *om,
     if (!BLE_MBUF_HDR_CRC_OK(hdr) ||
         BLE_MBUF_HDR_IGNORED(hdr) ||
         BLE_MBUF_HDR_AUX_INVALID(hdr) ||
-        (aux_data->flags_ll & BLE_LL_AUX_FLAG_SCAN_ERROR) ||
+        (aux_data && aux_data->flags_ll & BLE_LL_AUX_FLAG_SCAN_ERROR) ||
```
Comment (andrzej-kaczmarek) [inline nimble/controller/src/ble_ll_scan.c:3084]: aux_data cannot be null here, see comment at line 3075.
also please do not mix fixes for LL and host in single PR

Comment (KKopyscinski) [reply nimble/controller/src/ble_ll_scan.c]: ok, will remember :) so this is host-only PR now

File: nimble/host/mesh/src/testing.c [hunk 5]
```diff
@@ -171,11 +171,13 @@ void bt_test_print_credentials(void)
...
-		console_printf("\tNID: %02x\n", subnet->keys->msg.nid);
-		console_printf("\tEncKey: %s\n",
-			       bt_hex(subnet->keys->msg.enc, 16));
-		console_printf("\tPrivKey: %s\n",
-			       bt_hex(subnet->keys->msg.privacy, 16));
+		if (subnet) {
+			console_printf("\tNID: %02x\n", subnet->keys->msg.nid);
+			console_printf("\tEncKey: %s\n",
+						   bt_hex(subnet->keys->msg.enc, 16));
+            console_printf("\tPrivKey: %s\n",
```
Comment (michal-narajowski) [inline nimble/host/mesh/src/testing.c:178]: Fix whitespace here.

Comment (KKopyscinski) [reply nimble/host/mesh/src/testing.c]: done

**Review by michal-narajowski — APPROVED** [review]
LGTM

**Pull request merged**